### ISSUE TITLE
Buck blows up on backticks in kotlin

### ISSUE THREAD

**Issue by ZacSweers** [facebook/buck#1621]
If you have a backtick in an API for kotlin, buck will fail with a stacktrace like this.

Example: `fun 'in'()` (replace the `'` with backticks, github won't format it nicely)

```
./buckw test //libraries/foundation/realtime/realtime-projectgen/... --include unit                                                                                    
Buck encountered an internal error
java.lang.IllegalStateException
	at com.google.common.base.Preconditions.checkState(Preconditions.java:429)
	at com.facebook.buck.jvm.java.AccumulateClassNamesStep.parseClassHashes(AccumulateClassNamesStep.java:167)
	at com.facebook.buck.jvm.java.JavaLibraryRules.initializeFromDisk(JavaLibraryRules.java:81)
	at com.facebook.buck.jvm.java.DefaultJavaLibrary.initializeFromDisk(DefaultJavaLibrary.java:333)
	at com.facebook.buck.jvm.java.DefaultJavaLibrary.initializeFromDisk(DefaultJavaLibrary.java:1)
	at com.facebook.buck.rules.BuildOutputInitializer.initializeFromDisk(BuildOutputInitializer.java:47)
	at com.facebook.buck.rules.CachingBuildRuleBuilder.doInitializeFromDisk(CachingBuildRuleBuilder.java:1276)
	at com.facebook.buck.rules.CachingBuildRuleBuilder.finalizeBuildRule(CachingBuildRuleBuilder.java:403)
	at com.facebook.buck.rules.CachingBuildRuleBuilder.lambda$4(CachingBuildRuleBuilder.java:288)
	at com.facebook.buck.rules.CachingBuildRuleBuilder.lambda$21(CachingBuildRuleBuilder.java:1634)
	at com.google.common.util.concurrent.AbstractTransformFuture$AsyncTransformFuture.doTransform(AbstractTransformFuture.java:211)
	at com.google.common.util.concurrent.AbstractTransformFuture$AsyncTransformFuture.doTransform(AbstractTransformFuture.java:200)
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:130)
	at com.google.common.util.concurrent.MoreExecutors$5$1.run(MoreExecutors.java:952)
	at com.facebook.buck.util.concurrent.WeightedListeningExecutorService.lambda$2(WeightedListeningExecutorService.java:104)
	at com.facebook.buck.util.concurrent.WeightedListeningExecutorService.lambda$0(WeightedListeningExecutorService.java:78)
	at com.google.common.util.concurrent.AbstractTransformFuture$AsyncTransformFuture.doTransform(AbstractTransformFuture.java:211)
	at com.google.common.util.concurrent.AbstractTransformFuture$AsyncTransformFuture.doTransform(AbstractTransformFuture.java:200)
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:130)
	at com.google.common.util.concurrent.MoreExecutors$5$1.run(MoreExecutors.java:952)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)

    When building rule //libraries/foundation/realtime/realtime-projectgen:test_main#testsjar.
Downloaded 68.87 Kbytes/sec avg, 3 artifacts, 48.00 Kbytes
Building: finished in 2.1 sec (100%) 397/398 jobs, 1 updated, 0.3% cache miss
  Total time: 2.3 sec
```

This is kind of serious, as backticks in kotlin are quite normal to avoid keyword collisions

**Comment by hzsweers** [comment]
CC @kageiit

**Comment by noiz354** [comment]
i also face that issue, u can reproduce at this link 
https://github.com/noiz354/okbuck/tree/dbflow_illegalstateexception

is there any new update.

**Comment by hkocinneide** [comment]
I found this issue as well in some Android unit tests. I had some anonymous objects in my code, and removing either the backtick'd method name or the anonymous objects resolved the issue.

**Comment by kburovsc** [comment]
Are there any plans to get this fixed?

**Comment by hzsweers** [comment]
It likely would need to come as a community contribution like all the other kotlin support

**Comment by elucash** [comment]
_Adding description from the duplicate issue_

When there are any local/anonymous/lambda classes are used in Kotlin functions with names containing spaces 

```kotlin
fun `my test name with spaces`() {
  val j = object : Consumer<String> {
    override fun accept(t: String) {}
  }
}
```

```
Buck encountered an internal error

java.lang.IllegalStateException

 at com.google.common.base.Preconditions.checkState(Preconditions.java:485)
 at com.facebook.buck.jvm.java.AccumulateClassNamesStep.parseClassHashes(AccumulateClassNamesStep.java:167)
 at com.facebook.buck.jvm.java.JavaLibraryRules.initializeFromDisk(JavaLibraryRules.java:62)
 at com.facebook.buck.jvm.java.DefaultJavaLibrary.initializeFromDisk(DefaultJavaLibrary.java:375)
 at com.facebook.buck.jvm.java.DefaultJavaLibrary.initializeFromDisk(DefaultJavaLibrary.java:84)
 at com.facebook.buck.core.rules.attr.BuildOutputInitializer.initializeFromDisk(BuildOutputInitializer.java:57)
 at com.facebook.buck.core.build.engine.impl.CachingBuildRuleBuilder.doInitializeFromDisk(CachingBuildRuleBuilder.java:1304)
 at com.facebook.buck.core.build.engine.impl.CachingBuildRuleBuilder.finalizeBuildRule(CachingBuildRuleBuilder.java:549)
 at com.facebook.buck.core.build.engine.impl.CachingBuildRuleBuilder.lambda$build$4(CachingBuildRuleBuilder.java:412)
 at com.facebook.buck.core.build.engine.impl.CachingBuildRuleBuilder.lambda$ruleAsyncFunction$25(CachingBuildRuleBuilder.java:1346)
 at com.google.common.util.concurrent.AbstractTransformFuture$AsyncTransformFuture.doTransform(AbstractTransformFuture.java:206)
 at com.google.common.util.concurrent.AbstractTransformFuture$AsyncTransformFuture.doTransform(AbstractTransformFuture.java:195)
 at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:115)
 ... 10 more

    When building rule //some:test#testsjar.
```

Seems like checksum parser chokes on extra spaces in `<filename> <checksum>` lines. Maybe it would be better if the compiler would not emit classes with spaces and derive bytecode class names differently.

While checksum parsing can be fixed on the side of Buck. Do anyone know if there's a way to change Kotlinc to not use spaces is name mangled filenames for local classes etc?

**Comment by NickFirmani** [comment]
This was resolved by ef9a62adb3d888b97e97b5eda8c278c9b59ce9a7
